app/full_text_search.py

```diff
@@ -106,16 +109,78 @@ def search(repo, query_txt, max_results):
             results.append(result)
     return results
 
-def create_document(**kwargs):
+
+def create_jp_name_document(**kwargs):
+    """
+    Creates document(romanized jp name data) for full text search.
+    It should be called in create_document method.
+    """
+    fields = []
+    romanized_names_list = []
+    for field in kwargs:
+        romanized_japanese_name =\
+                script_variant.romanize_japanese_name_by_name_dict(kwargs[field])
+        romanized_names_list.append(romanized_japanese_name)
+        fields.append(
+            appengine_search.TextField(name=field+'_romanized_by_jp_name_dict',
+                                       value=romanized_japanese_name)
+        )
+    romanized_jp_names =\
+            ':'.join([name for name in romanized_names_list if name])
+    fields.append(appengine_search.TextField(name='romanized_jp_names',
+                                             value=romanized_jp_names))
+    return fields
+
+
+def create_jp_location_document(**kwargs):
+    """
+    Creates document(romanized jp location data) for full text search.
+    It should be called in create_document method.
+    """
+    fields = []
+    for field in kwargs:
+        romanized_japanese_location =\
+                script_variant.romanize_japanese_location(kwargs[field])
+        fields.append(
+            appengine_search.TextField(name=field+'_romanized_by_jp_location_dict',
+                                       value=romanized_japanese_location)
+        )
+    return fields
+
+def create_document(record_id, repo, **kwargs):
     """
     Creates document for full text search.
-    It should be called in add_record_to_index method.model.Person.get_by_key_name(repo + ':' + id
+    It should be called in add_record_to_index method.
     """
-    doc_id = kwargs['repo'] + ':' + kwargs['record_id']
     fields = []
+
+    # Add repo and record_id to fields
+    doc_id = repo + ':' + record_id
+    fields.append(appengine_search.TextField(name='repo', value=repo))
+    fields.append(appengine_search.TextField(name='record_id', value=record_id))
+
+    # Add name and location romanized by unidecode
     for field in kwargs:
+        romanized_value = script_variant.romanize_word(kwargs[field])
         fields.append(
-            appengine_search.TextField(name=field, value=kwargs[field]))
+            appengine_search.TextField(name=field, value=romanized_value))
+
+    # Add name romanized by japanese name dictionary
+    fields.extend(create_jp_name_document(
```

Comment:
Is the function name misleading? It's not returning a document by a list of fields.